File: src/main/java/fi/finlit/edith/sql/domain/NoteComment.java
package fi.finlit.edith.sql.domain;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import org.hibernate.annotations.Type;
import org.joda.time.DateTime;

@Entity
@Table(name = "notecomment")
public class NoteComment {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne
    private Note note;

    private String message;

    private String username;

    @Type(type="org.joda.time.contrib.hibernate.PersistentDateTime")
    private DateTime createdAt;

    public NoteComment() { }

    public NoteComment(Note note, String message, String username) {
        this.note = note;
        this.message = message;
        this.username = username;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getMessage() {
        return message;
    }

    public Note getNote() {
        return note;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public void setNote(Note note) {
        this.note = note;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getUsername() {
        return username;
    }

    public DateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(DateTime createdAt) {
        this.createdAt = createdAt;
    }
}

File: src/main/scala/fi/finlit/edith/sql/domain/NoteComment.scala
package fi.finlit.edith.sql.domain

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.ManyToOne
import javax.persistence.Table
import org.hibernate.annotations.Type
import org.joda.time.DateTime
import fi.finlit.edith.Identifiable
import scala.reflect.BeanProperty
//remove if not needed
import scala.collection.JavaConversions._

@Entity
@Table(name = "notecomment")
class NoteComment() extends Identifiable {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  @BeanProperty
  var id: java.lang.Long = _

  @ManyToOne
  @BeanProperty
  var note: Note = _

  @BeanProperty
  var message: String = _

  @BeanProperty
  var username: String = _

  @Type(type = "org.joda.time.contrib.hibernate.PersistentDateTime")
  @BeanProperty
  var createdAt: DateTime = _

  def this(note: Note, message: String, username: String) {
    this.note = note
    this.message = message
    this.username = username
  }
}
